### PR TITLE
Add transaction capabilities to bindings and fix some issues

### LINKED ISSUE #1499: Python: add transaction capabilities

### LINKED ISSUE #1500: Nodejs: add transaction capabilities

### PR DESCRIPTION
**Pull request by DaughterOfMars**
# Description of change

Adds the transaction capabilities flags to the nodejs and python bindings.

NOTE: python still uses transaction essence so I just put it there for now.

## Links to any relevant issues

Closes #1499 
Closes #1500

### CODE REVIEW

File: bindings/python/iota_sdk/types/address.py [hunk 1]
```diff
@@ -105,12 +105,12 @@ class RestrictedAddress:
...
 
-    def with_allowed_capabilities(self, allowed_capabilities: bytes):
+    def with_allowed_capabilities(self, capabilities: bytes):
         """Sets the allowed capabilities from a byte array.
         Attributes:
-            allowed_capabilities: The allowed capabilities bitflags.
+            capabilities: The allowed capabilities bitflags.
         """
-        self.allowed_capabilities = '0x00' + allowed_capabilities.hex()
+        self.allowed_capabilities = hex(len(capabilities)) + capabilities.hex()
```
Comment (thibault-martinez) [inline bindings/python/iota_sdk/types/address.py]: I'm not fully sure the length is required?

Comment (DaughterOfMars) [reply bindings/python/iota_sdk/types/address.py]: why not?

Comment (thibault-martinez) [reply bindings/python/iota_sdk/types/address.py]: We just don't put lengths in hex, I've confirmed with Phillip that it should be without length

Comment (thibault-martinez) [reply bindings/python/iota_sdk/types/address.py]: There is no real need for it, it's parseable without

Comment (DaughterOfMars) [reply bindings/python/iota_sdk/types/address.py]: ohhhh duh

File: bindings/python/iota_sdk/types/essence.py [hunk 2]
```diff
@@ -50,10 +51,18 @@ class RegularTransactionEssence:
...
     context_inputs: Optional[List[ContextInput]] = None
     allotments: Optional[List[ManaAllotment]] = None
+    capabilities: HexStr = field(default='0x', init=False)
     payload: Optional[Payload] = None
     type: int = field(
         default_factory=lambda: EssenceType.RegularTransactionEssence,
         init=False)
 
+    def with_capabilities(self, capabilities: bytes):
+        """Sets the transaction capabilities from a byte array.
```
Comment (Thoralf-M) [inline bindings/python/iota_sdk/types/essence.py:61]: There is also a `bytearray` type, comment or parameter should be aligned
`The difference between bytes() and bytearray() is that bytes() returns an object that cannot be modified, and bytearray() returns an object that can be modified.`

Comment (DaughterOfMars) [reply bindings/python/iota_sdk/types/essence.py:61]: I mean...both are arrays

Comment (Thoralf-M) [reply bindings/python/iota_sdk/types/essence.py:61]: True, nvm

**Review by Alex6323 — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**